packages/aws-cdk-lib/aws-ssm/test/parameter.test.ts

```diff
@@ -963,3 +963,40 @@ test('fails if parameterName is undefined and simpleName is "false"', () => {
   // THEN
   expect(() => new ssm.StringParameter(stack, 'p', { simpleName: false, stringValue: 'foo' })).toThrow(/If "parameterName" is not explicitly defined, "simpleName" must be "true" or undefined since auto-generated parameter names always have simple names/);
 });
+
```

Comment:
Can you add a test here that uses a lazy value? A good example would be a cross env bucketname.

```ts
const bucket = new s3.Bucket(this, 'Bucket', {
  bucketName: cdk.PhysicalName.GENERATE_IF_NEEDED,
});

const stack = new cdk.Stack(scope, 'Cross', {
  env: { region: 'us-east-1', account: process.env.CDK_DEFAULT_ACCOUNT },
});
ssm.StringParameter.fromStringParameterAttributes(stack, 'param', {
  parameterName: bucket.bucketName,
  simpleName: true,
});
```

Reply:
Note: I used iam.Role instead as i don't know if it is allowed to add a new dependency on other modules.